src/corehost/common/pal.unix.cpp

```diff
@@ -112,11 +124,64 @@ bool pal::is_path_rooted(const pal::string_t& path)
     return path.front() == '/';
 }
 
+bool pal::get_default_breadcrumb_store(string_t* recv)
+{
+    recv->clear();
+    pal::string_t ext;
+    if (!get_default_extensions_directory(&ext))
+    {
+        trace::info(_X("Breadcrumb store not obtained because dotnet extensions could not be obtained"));
+        return false;
+    }
+
+    append_path(&ext, _X("breadcrumbs"));
+    if (!pal::directory_exists(ext))
+    {
+        trace::info(_X("Breadcrumb store does not exist in [%s]"), ext.c_str());
+        return false;
+    }
+
+    if (access(ext.c_str(), (R_OK | W_OK)) != 0)
```

Comment:
@ericstj, I haven't addressed your PR yet, as I wanted to test this out first. Will send another commit for this one specifically.